Light Vox Engine/Graphics/FrameResource.h

```diff
@@ -76,34 +87,55 @@ class FrameResource
         Camera* camera
     );
 
-    //culmination of all of the command lists
-    ID3D12CommandList* batchedCommandList[ 1 + LV_COMMAND_LIST_COUNT ];
-
-    //command list/allocator for the stages (pre, mid, post)
-    Microsoft::WRL::ComPtr<ID3D12CommandAllocator> commandAllocators[ LV_COMMAND_LIST_COUNT ];
+    //a batched command list for g-buffer stuff and scene stuff
+    ID3D12CommandList* batchedCommandList[ LV_NUM_CONTEXTS * 1 + LV_COMMAND_LIST_COUNT ];
+    Microsoft::WRL::ComPtr<ID3D12GraphicsCommandList> geometryCmdLists[ LV_NUM_CONTEXTS ];
     Microsoft::WRL::ComPtr<ID3D12GraphicsCommandList> commandLists[ LV_COMMAND_LIST_COUNT ];
 
     //command list/allocator for 'per scene' stuff (split between cores)
-    Microsoft::WRL::ComPtr<ID3D12CommandAllocator> sceneCommandAllocators[ LV_NUM_CONTEXTS ];
-    Microsoft::WRL::ComPtr<ID3D12GraphicsCommandList> sceneCommandLists[ LV_NUM_CONTEXTS ];
+#ifdef _DEBUG
+    Microsoft::WRL::ComPtr<ID3D12GraphicsCommandList> debugCmdLists[ LV_NUM_CONTEXTS ];
+#endif
 
     //used to synchronize instructions
     UINT64 fenceValue;
 private:
+    Microsoft::WRL::ComPtr<ID3D12CommandAllocator> geometryCmdAllocators[ LV_NUM_CONTEXTS ];
+    Microsoft::WRL::ComPtr<ID3D12CommandAllocator> commandAllocators[ LV_COMMAND_LIST_COUNT ];
+
+#ifdef _DEBUG
+    Microsoft::WRL::ComPtr<ID3D12CommandAllocator> debugCommandAllocator;
+#endif
+
+    inline void InitCmdAllocatorsAndLists( ID3D12Device * device, UINT frameResourceIndex );
+    inline void InitDescriptorHandles( ID3D12Device * device, ID3D12DescriptorHeap * dsvHeap,
+        ID3D12DescriptorHeap * rtvHeap, ID3D12DescriptorHeap * cbvSrvHeap, UINT frameResourceIndex );
+    inline void InitGraphicsResources( ID3D12Device * device, ID3D12DescriptorHeap * dsvHeap, ID3D12DescriptorHeap * rtvHeap,
+        ID3D12DescriptorHeap * cbvSrvHeap, D3D12_VIEWPORT * viewport, UINT frameResourceIndex );
+    inline void InitCBV( ID3D12Device * device, ID3D12DescriptorHeap * cbvSrvHeap, UINT frameResourceIndex );
+
     //pointer to things we need to properly render
-    Microsoft::WRL::ComPtr<ID3D12PipelineState> pso;
+    Microsoft::WRL::ComPtr<ID3D12PipelineState> geometryBufferPso;
+    Microsoft::WRL::ComPtr<ID3D12PipelineState> deferredPso;
 
     //the gpu side of the scene constant buffer
     Microsoft::WRL::ComPtr<ID3D12Resource> sceneConstantBuffer;
-
-    //instance upload buffer
     Microsoft::WRL::ComPtr<ID3D12Resource> instanceUploadBuffer;
+    //Microsoft::WRL::ComPtr<ID3D12Resource> lightConstantBuffer;
```

Comment:
remove this comment if we don't need it

Reply:
We'll need lights eventually so I'll keep it there for now